### DISCUSSION TITLE
feeling about using sumatrapdf

### DISCUSSION THREAD

**Discussion by AHK0** [sumatrapdfreader/sumatrapdf#3338] [General]
have recently started to use this software, and have a complex mixed feeling of the user experience.

have used >10 different document viewer (mainly pdf) softwares, including the more popular foxit, xchange and many other products. there are some apparent pros and cons with sumatrapdf.
pros: a very nice and amazing software, simple, fast, usuage-low, etc. many people have similar feelings, and it is not necessary to list too much.
there are also some apparent cons. regarding the current software design, i think it could be well improved.
1. the portable is not 'portable'. for safety reasons, sumatra is designed as not portable when located in the 'program files'. for many people in the world including myself, they usually throw portable softwares simply into the system program files folder. this folder does not mean unsafe at all. actually there are hundreds of other portable softwares which work well under the "program files", including the portable versions of foxit reader, x-change viewer, etc (http://portableapps.com/).
2. button icons in the toolbar could be offered with at least small, medium and large options, and toolbar should be able to be customized with the icons and their sizes.
3. the way it works for annotations and captures is so not simple (opposite to the "simple" of the software itself). for other viewers, people only need to click once the annotation or capture button in the toolbar, and then they can annotate any text and capture any regions with a simple clink. for sumatrapdf, these are quite complex, because you have to each time click several times (and you even need to use shortcuts). about the capture, you may say there are other capture softwares --- which is certainly true but only makes the operations complex. because in similar softwares, just one click is enough for capturing any regions of text, picture or text + picture as image and exporting it as image files/copying it to clipboard.
4. the complex operations of software (shortcut frequently) make it unsuitable for touch-screen computers such as MS surface. this should definitely be considered, concerning the button icon sizes, toolbar sizes, buttons of annotation and capture in the toolbar, and many others. even for non touch-screen computers, the operations should also be simple.

in brief, pure viewer does not have much living space, and this may be the reason the developer have added annotation and capture functions to the sumatrapdf. however, the involved operations should be similarly simple as the "simple" of the software itself. keyboard shortcuts should not be too much concerned, because >95% (maybe >99%) of the users (market) only want to simply use mouse (or screen-touch if available). this is what a simple may indeed mean, and would make the software more popular.

**Comment by GitHubRulesOK** [comment]
For portable apps have you not considered where they hold their registry and runtime data?
```
[PortableApps.comLauncher]
PluginsDir=C:\Users\me\AppData\Local\Temp\nsk71E8.tmp

[Launch]
ProgramExecutable=PDFXEdit\PDFXEdit.exe
DirectoryMoveOK=yes
SupportsUNC=yes

[FilesMove]
settings.dat=%PAL:AppDir%\PDFXEdit
history.dat=%PAL:AppDir%\PDFXEdit
```
the difference is they slowly load and unload the registry and remote support folders at runtime. There is NO SUCH thing as a stealth portable app and I have checked thousands, the MS system security keeps track of all events and can be often be forensically reversed to show data history. MS  positively discourage exploitable data in the secured  areas.

All those desired features should be available from the Oriental full time team of many developers and resellers that rip off SumatraPDF and charge $100 per copy , or by subscription from the Adobe team composed of 10s of thousands full time staff, but then those commercial whistles and bells seem to slow those versions down considerably. I know because I have paid for those slower editing abilities as required for heavy PDF edits.

**Reply by AHK0** [reply]
anyway, this is not complaining. the reason why the users are suggesting is because they love this software and want it to be more successful. this is the key point!
there are indeed many amazing open-source free softwares, of cource sumatra is one of them.

if you are talking about the money, you could make this sumatra commercial -- the users who really like it will pay for it. to me, if it is good enough, i can also pay for it. but, this is not the point what i have written that long for the suggestions, because i strongly feel that sumatra could be more simple and be more successful.

the portable you mentioned is acceptable to the world (if i set sumatrapdf as the default pdf viewer, it (the portable version) will write to registry, but this is fine to me.

your responses suriprise me, i will stop writing anything for this software, byebye.



> For portable apps have you not considered where they hold their registry and runtime data?
> 
> ```
> [PortableApps.comLauncher]
> PluginsDir=C:\Users\me\AppData\Local\Temp\nsk71E8.tmp
> 
> [Launch]
> ProgramExecutable=PDFXEdit\PDFXEdit.exe
> DirectoryMoveOK=yes
> SupportsUNC=yes
> 
> [FilesMove]
> settings.dat=%PAL:AppDir%\PDFXEdit
> history.dat=%PAL:AppDir%\PDFXEdit
> ```
> 
> the difference is they slowly load and unload the registry and remote support folders at runtime. There is NO SUCH thing as a stealth portable app and I have checked thousands, the MS system security keeps track of all events and can be often be forensically reversed to show data history. MS positively discourage exploitable data in the secured areas.
> 
> All those desired features should be available from the Oriental full time team of many developers and resellers that rip off SumatraPDF and charge $100 per copy , or by subscription from the Adobe team composed of 10s of thousands full time staff, but then those commercial whistles and bells seem to slow those versions down considerably. I know because I have paid for those slower editing abilities as required for heavy PDF edits.

**Comment by AnxiousPleaseHelp** [comment]
> have used >10 different document viewer (mainly pdf) softwares, including the more popular foxit, xchange and many other products. there are some apparent pros and cons with sumatrapdf. pros: a very nice and amazing software, simple, fast, usuage-low, etc. many people have similar feelings, and it is not necessary to list too much. there are also some apparent cons. regarding the current software design, i think it could be well improved.
> 
>     1. the portable is not 'portable'. for safety reasons, sumatra is designed as not portable when located in the 'program files'. for many people in the world including myself, they usually throw portable softwares simply into the system program files folder. this folder does not mean unsafe at all. actually there are hundreds of other portable softwares which work well under the "program files", including the portable versions of foxit reader, x-change viewer, etc (http://portableapps.com/).

I have a similar issue to you. I know you originally posted this in 2023 so I hope you were able to find a solution!

This is my thread here: https://github.com/sumatrapdfreader/sumatrapdf/discussions/5027
